doc.go

```diff
@@ -22,6 +22,21 @@ of sub-packages.
 All clients in sub-packages are configurable via client options. These options are
 described here: https://godoc.org/google.golang.org/api/option.
 
+## Endpoint
+
+Endpoint configuration is used to specify the API endpoint to which requests are
+sent. It is primarily used for services that support or require regional endpoints.
+For example, the Vertex AI service recommends that you set the location with the
+features you want that is closest to your physical location or the physical location
+of your users. There is no global endpoint for Vertex AI. See
+[Vertex AI - Locations](https://cloud.google.com/vertex-ai/docs/general/locations)
+for more details. The following example demonstrates configuring a Vertex AI client
+with a regional endpoint:
+
+	ctx := context.Background()
+	endpoint := "us-central1-aiplatform.googleapis.com"
+	client, err := aiplatform.NewDatasetClient(ctx, option.WithEndpoint(endpoint))
+
```

Comment:
The other use case where I see people using endpoint overrides is when they are testing against fake servers. Do you think we should mention this at all here as another potential use-case? There is an example of how this is done here: https://github.com/googleapis/google-cloud-go/blob/main/testing.md#testing-grpc-services-using-fakes

Reply:
Added, thanks!